Crash on DigitalOcean droplet

Tool crashes on DigitalOcean droplet (Shared CPU, 2 vCPUs, 4 GB, 25 GB, 4 TB). Startup method: docker install.

goroutine 44381 [running]:
math/rand.(*Rand).Intn(0x80acea, 0x46817b)
        /usr/local/go/src/math/rand/rand.go:168 +0x65
math/rand.Intn(...)
        /usr/local/go/src/math/rand/rand.go:337
main.httpJob.func1(0xc003364d00)
        /usr/src/app/main.go:229 +0x45
net/http.(*Transport).connectMethodForRequest(0xc00d50ab40, 0xc00d23d2c0)
        /usr/local/go/src/net/http/transport.go:834 +0x91
net/http.(*Transport).roundTrip(0xc00d50ab40, 0xc003364d00)
        /usr/local/go/src/net/http/transport.go:571 +0x6c5
net/http.(*Transport).RoundTrip(0xc003364d00, 0x8d2480)
        /usr/local/go/src/net/http/roundtrip.go:18 +0x19
net/http.send(0xc003364c00, {0x8d2480, 0xc00d50ab40}, {0x7fe120, 0x48a501, 0xbb6680})
        /usr/local/go/src/net/http/client.go:252 +0x5d8
net/http.(*Client).send(0xc00e272c00, 0xc003364c00, {0xc00e273c20, 0x2f, 0xbb6680})
        /usr/local/go/src/net/http/client.go:176 +0x9b
net/http.(*Client).do(0xc00e272c00, 0xc003364c00)
        /usr/local/go/src/net/http/client.go:725 +0x908
net/http.(*Client).Do(...)
        /usr/local/go/src/net/http/client.go:593
main.httpJob({0x8db078, 0xc002cb5b80}, 0xc0000a6310, {0xc00230e1e0, 0x9a, 0xa0})
        /usr/src/app/main.go:282 +0xa9e
created by main.main
        /usr/src/app/main.go:586 +0x725

P.S. In Droplets with worse characteristics tool is killed by OOM-killer.

I've checked proposed solution but it didn't help. Swap size range I used: 600 MB - 8 GB

panic: invalid argument to Intn                                                                               

goroutine 30540 [running]:
math/rand.(*Rand).Intn(0x80acea, 0x46817b)
        /usr/local/go/src/math/rand/rand.go:168 +0x65
math/rand.Intn(...)
        /usr/local/go/src/math/rand/rand.go:337
main.httpJob.func1(0xc000dde500)
        /usr/src/app/main.go:229 +0x45
net/http.(*Transport).connectMethodForRequest(0xc001669040, 0xc00312c600)
        /usr/local/go/src/net/http/transport.go:834 +0x91
net/http.(*Transport).roundTrip(0xc001669040, 0xc000dde400)
        /usr/local/go/src/net/http/transport.go:571 +0x6c5
net/http.(*Transport).RoundTrip(0xc000dde400, 0x8d2480)
        /usr/local/go/src/net/http/roundtrip.go:18 +0x19
net/http.send(0xc000dde300, {0x8d2480, 0xc001669040}, {0x7fe120, 0x48a501, 0xbb6680})
        /usr/local/go/src/net/http/client.go:252 +0x5d8
net/http.(*Client).send(0xc001d07920, 0xc000dde300, {0xc00312c500, 0x8d26e0, 0xbb6680})
        /usr/local/go/src/net/http/client.go:176 +0x9b
net/http.(*Client).do(0xc001d07920, 0xc000dde300)
        /usr/local/go/src/net/http/client.go:725 +0x908
net/http.(*Client).Do(...)
        /usr/local/go/src/net/http/client.go:593
main.httpJob({0x8db078, 0xc002f05300}, 0xc0000a6310, {0xc002ecc200, 0x1e6, 0x200})
        /usr/src/app/main.go:282 +0xa9e
created by main.main
        /usr/src/app/main.go:586 +0x725

--- a/main.go
+++ b/main.go
@@ -227,6 +227,9 @@ func httpJob(ctx context.Context, l *logs.Logger, args JobArgs) error {
 			l.Debug("proxyURLs: %v", proxyURLs)
 			// Return random proxy from the list
 			proxy = func(r *http.Request) (*url.URL, error) {
+				if len(proxyURLs) == 0 {
+					return nil, fmt.Errorf("proxylist is empty")
+				}
 				proxyID := rand.Intn(len(proxyURLs))
 				proxyString := proxyURLs[proxyID]
 				u, err := url.Parse(proxyString)

